plugins/outputs/mqtt/mqtt.go

```diff
@@ -58,6 +59,20 @@ func (*MQTT) SampleConfig() string {
 	return sampleConfig
 }
 
+func (m *MQTT) Init() error {
+	var err error
+	m.generator, err = NewTopicNameGenerator(m.TopicPrefix, m.Topic)
+	if err != nil {
+		return err
+	}
+	for _, p := range strings.Split(m.Topic, "/") {
+		if strings.ContainsAny(p, "#+") {
+			return fmt.Errorf("found forbidden character %s in the topic name %s", p, m.Topic)
+		}
+	}
```

Comment:
Can you please also move this code to `NewTopicNameGenerator`!

Reply:
Sure!